grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -169,4 +173,94 @@ class CycleCountService {
 
         return csv
     }
+
+    /**
+     * Batch method to start a cycle count.
+     * Fetch associated Cycle count with the CycleCountRequest and do the following:
+     * if a cycle count doesn't exist, create it, fetch all bin locations + inventory items for the product in a request,
+     * get the product availability for each item, filter out items that have qoh = 0 (but keep those that have qoh = 0 and were included in the previous count)
+     *
+     * */
+    List<CycleCountDto> startCycleCount(CycleCountStartBatchCommand command) {
+        List<CycleCountDto> cycleCounts = []
+        command.requests.each { CycleCountStartCommand request ->
+            CycleCount cycleCount = request.cycleCountRequest.cycleCount
+            if (!cycleCount) {
+                CycleCountDto cycleCountDto = createCycleCount(request, command.facility)
+                cycleCounts.add(cycleCountDto)
+                return
+            }
+            List<CycleCountItem> cycleCountItems = CycleCountItem.findAllByCycleCount(cycleCount)
+            CycleCountDto cycleCountDto = updateCycleCount(cycleCount, cycleCountItems, request)
+            cycleCounts.add(cycleCountDto)
+        }
+        return cycleCounts
+    }
+
+    CycleCountDto createCycleCount(CycleCountStartCommand request, Location facility) {
+        CycleCount newCycleCount = new CycleCount(
+                facility: facility,
+                dateLastRefreshed: new Date()
+        )
+        List<AvailableItem> itemsToSave = determineCycleCountItemsToSave(facility, request.cycleCountRequest.product)
+        newCycleCount.save()
+        // 1:1 association between cycle count and cycle count request
+        request.cycleCountRequest.cycleCount = newCycleCount
```

Comment:
since the cycle count request's status is a persistent value, I'm wondering if we should change it right here, if so - should it be REQUESTED or IN_PROGRESS?